### PR TITLE
Typing indicators

### LINKED ISSUE #124: [FEATURE] typing indicators
### Related Pull Request on other codebase:
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12015
fixes for that:
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/13559
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/13698 (we might have the parts of this that don't relate to tying indicators)
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/13727
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/14809

or https://github.com/Whitesands13/Whitesands/pull/288 but fix it because it did not work in this PR

### Additional Info:

they are extremely fucking useful for RP

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Resolves: #124

Feature requested by @tiramisuapimancer 

image

It's rebindable, so if you don't want to show it then just rebind T or whatever your chat key is to normal say

Port of: Citadel-Station-13/Citadel-Station-13#12015 through Whitesands13/Whitesands#288

## Why It's Good For The Game
Allows people to tell if you're typing a message, so they don't run away.

## Changelog
:cl:
add: Typing indicators. They're rebindable so if you don't want it just rebind T to normal say
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_say.dm [hunk 1]
```diff
@@ -10,6 +10,15 @@
 	if(message)
 		say(message)
 
+///The say verb, but with typing indicator shown
+/mob/verb/say_with_indicator()
+	set name = "Say_indicator"
```
Comment (ZephyrTFA) [inline code/modules/mob/mob_say.dm]:
```suggestion
	set name = "Say Visible"
```

File: code/modules/mob/mob_say.dm [hunk 2]
```diff
@@ -37,6 +46,15 @@
 
 	usr.emote("me",1,message,TRUE)
 
+///The me/emote verb, but with typing indicator shown
+/mob/verb/me_with_indicator()
+	set name = "Me_indicator"
```
Comment (ZephyrTFA) [inline code/modules/mob/mob_say.dm]:
```suggestion
	set name = "Me Visible"
```

**Comment by Azarak** [comment]
An important thing to keep in mind, the verb wrappers are synchronous and will not immediately make a chat box popup, unlike the current asynchronous verbs, it will have to wait for a server ping-pong. This gets really annoying even with low latencies.

**Comment by Azarak** [comment]
I could PR an asynchronous alternative if that's desired, keep in mind it will be slightly hacky, as there is no other way

**Pull request closed without merging**